### PR TITLE
Add us-east-2

### PR DESCRIPTION
**Pull request by nckturner**
* add supported region us-east-2
* add IsSupportedRegion() helper

### Description
Please explain the changes you made here.

### Checklist
- [x] Code compiles correctly (i.e `make build`)
- [ ] Added tests that cover your change (if possible)
- [x] All tests passing (i.e. `make test`)
- [ ] Added/modified documentation as required (such as the README)
- [x] Added yourself to the `humans.txt` file

### CODE REVIEW

File: pkg/ctl/create/cluster.go [hunk 1]
```diff
@@ -101,8 +101,8 @@ func createClusterCmd() *cobra.Command {
 func doCreateCluster(cfg *api.ClusterConfig, ng *api.NodeGroup, name string) error {
 	ctl := eks.New(cfg)
 
-	if cfg.Region != api.EKSRegionUSWest2 && cfg.Region != api.EKSRegionUSEast1 && cfg.Region != api.EKSRegionEUWest1 {
-		return fmt.Errorf("%s is not supported only %s, %s and %s are supported", cfg.Region, api.EKSRegionUSWest2, api.EKSRegionUSEast1, api.EKSRegionEUWest1)
+	if !api.IsSupportedRegion(cfg.Region) {
+		return fmt.Errorf("%s is not supported. Only %v are supported", cfg.Region, api.SupportedRegions)
```
Comment (errordeveloper) [inline pkg/ctl/create/cluster.go]: how about this?
```suggestion
		return fmt.Errorf("%s is not supported (use one of: %s)", cfg.Region, strings.Join(api.SupportedRegions, ","))
```

File: pkg/eks/api/api.go [hunk 2]
```diff
@@ -30,9 +33,20 @@ const (
 var SupportedRegions = []string{
 	EKSRegionUSWest2,
 	EKSRegionUSEast1,
+	EKSRegionUSEast2,
 	EKSRegionEUWest1,
 }
 
+// IsSupportedRegion returns true if EKS is available in the region
+func IsSupportedRegion(region string) bool {
```
Comment (errordeveloper) [inline pkg/eks/api/api.go]: We have this already: https://github.com/weaveworks/eksctl/blob/0a87749aca78b5a94541651eb95bc4b41e4f2914/pkg/eks/api.go#L125-L133

File: pkg/ctl/create/cluster.go [hunk 3]
```diff
@@ -101,8 +101,8 @@ func createClusterCmd() *cobra.Command {
 func doCreateCluster(cfg *api.ClusterConfig, ng *api.NodeGroup, name string) error {
 	ctl := eks.New(cfg)
 
-	if cfg.Region != api.EKSRegionUSWest2 && cfg.Region != api.EKSRegionUSEast1 && cfg.Region != api.EKSRegionEUWest1 {
-		return fmt.Errorf("%s is not supported only %s, %s and %s are supported", cfg.Region, api.EKSRegionUSWest2, api.EKSRegionUSEast1, api.EKSRegionEUWest1)
+	if !api.IsSupportedRegion(cfg.Region) {
```
Comment (errordeveloper) [inline pkg/ctl/create/cluster.go]:
```suggestion
	if !ctl.IsSupportedRegion(cfg.Region) {
```

**Comment by errordeveloper** [comment]
@nckturner thanks for the PR anyway, are you at KubeCon this week?

**Comment by nckturner** [comment]
Whoops, how did I miss that?  Yes I am 👍

**Pull request closed without merging**